code/modules/projectiles/projectile.dm

```diff
@@ -950,29 +1155,50 @@
  * - deviation: (Optional) How the trajectory should deviate from the target in degrees.
  *   - //Spread is FORCED!
  */
-/obj/projectile/proc/preparePixelProjectile(atom/target, atom/source, list/modifiers = null, deviation = 0)
+/obj/projectile/proc/aim_projectile(atom/target, atom/source, list/modifiers = null, deviation = 0)
 	if(!(isnull(modifiers) || islist(modifiers)))
 		stack_trace("WARNING: Projectile [type] fired with non-list modifiers, likely was passed click params.")
 		modifiers = null
 
 	var/turf/source_loc = get_turf(source)
 	var/turf/target_loc = get_turf(target)
+
 	if(isnull(source_loc))
 		stack_trace("WARNING: Projectile [type] fired from nullspace.")
 		qdel(src)
 		return FALSE
 
-	trajectory_ignore_forcemove = TRUE
-	forceMove(source_loc)
-	trajectory_ignore_forcemove = FALSE
+	if(fired)
+		stack_trace("WARNING: Projectile [type] was aimed after already being fired.")
+		qdel(src)
+		return FALSE
 
+	free_hitscan_forceMove = TRUE
+	forceMove(source_loc)
 	starting = source_loc
 	pixel_x = source.pixel_x
 	pixel_y = source.pixel_y
 	original = target
+
+	// Trim off excess pixel_x/y by converting them into turf offset
+	if (abs(pixel_x) > ICON_SIZE_X * 0.5)
+		for (var/i in 1 to floor(abs(pixel_x) + ICON_SIZE_X * 0.5) / ICON_SIZE_X)
+			var/turf/new_loc = get_step(source_loc, pixel_x > 0 ? EAST : WEST)
+			if (!istype(new_loc))
+				break
+			source_loc = new_loc
+		pixel_x = pixel_x % (ICON_SIZE_X * 0.5)
+
```

Comment:
the for loop predicate is real verbose but a while doesn't quite fit here cause you'd need a finishing modulus anyway, can't decide what's ebst

Reply:
m pretty sure this is the best tbh